src/popover/container.tsx

```diff
@@ -148,11 +156,15 @@ export default function PopoverContainer({
 
       // Allow popover body to scroll if can't fit the popover into the container/viewport otherwise.
       if (scrollable) {
-        body.style.maxHeight = boundingOffset.height + 'px';
+        body.style.maxHeight = boundingBox.height + 'px';
         body.style.overflowX = 'hidden';
         body.style.overflowY = 'auto';
       }
 
+      if (allowVerticalOverflow) {
+        scrollRectangleIntoView(boundingBox);
+      }
```

Comment:
I do not understand the logic around this scroll behavior. Let's try together

1. First, we try all possible positions. If we can fit popover fully, exit with the chosen position
2. If we could not fit, we try to find best possible option, giving preference to positions with more available height
3. Limit the popover height and add overflow
4. Update page scroll (this particular line). At this point, all offsets calculated on step 2 are not valid anymore, because elements position changed.

Should the step 4 happen somewhere else in the workflow, or not happen at all?

Reply:
Your understanding is quite correct, and it's true that the top position value will be off sync for one frame. The scrolling itself then triggers a call to the [positionHandler](https://github.com/cloudscape-design/components/blob/04597474026edf3fb2fd665797dffc96d8ca57d9/src/popover/container.tsx#L175C7-L175C7), which corrects this because it recalculates the position relative to the new position of the trigger. I can't noticeable reproduce the initial misplacement (even with CPU throttling) but I will try to change this if it doesn't get too complicated.

Reply:
What if we move the step 4 after 1? Is there a chance that normal position will fit after scrolling?

Reply:
Handled this [here](bff2287b713bd775cfde3144183552067e52b381) by giving the popover the after-scroll position already right before (eventually) scrolling.